src/vs/workbench/contrib/terminal/browser/links/terminalWordLinkProvider.ts

```diff
@@ -68,14 +68,23 @@ export class TerminalWordLinkProvider extends TerminalBaseLinkProvider {
 			text += chars;
 		}
 
-		// Add the final link if there is one
+		// Add the final link if there is one and also remove trailing colon if exists
 		if (startX !== -1) {
-			result.push(new TerminalLink({ start: { x: startX + 1, y }, end: { x: line.length, y } }, text, this._xterm.buffer.active.viewportY, activateCallback, this._tooltipCallback, false, localize('searchWorkspace', 'Search workspace'), this._configurationService));
+			result.push(new TerminalLink({ start: { x: startX + 1, y }, end: { x: line.length, y } }, this._remove_trailing_colon(text), this._xterm.buffer.active.viewportY, activateCallback, this._tooltipCallback, false, localize('searchWorkspace', 'Search workspace'), this._configurationService));
 		}
 
 		return result;
 	}
 
+	// Remove trailing colon in the provided text
+	private _remove_trailing_colon(text: string): string {
+		if (text.length > 0 && text[text.length - 1] === ':') {
+			return text.slice(0, text.length - 1);
+		}
+		return text;
```

Comment:
since you're sure that : 
1) text won't be NULL.
2) case of trailing spaces isn't exist or no need to remove it.
so you don't need to check if length of the text is bigger than 0, i think TS has no problem to call slice on ZERO length string, you just need to check if trailing colon needed to be removed - i'd like to suggest TRIM instead of REMOVE in the function name but it's not necessary - 
```suggestion
		return (text.slice(-1) === ':')? text.slice(0,-1) : text;
```

Reply:
Thanks for the suggestion. Have incorporated these in my latest commit :)